worker/worker.go

```diff
@@ -283,7 +283,11 @@ func New(
 	taskList string,
 	options Options,
 ) Worker {
-	return internal.NewWorker(service, domain, taskList, options)
+	w, err := internal.NewWorker(service, domain, taskList, options)
+	if err != nil {
+		panic(err)
```

Comment:
hm :\

might be worth deferring error handling until it's started?  or does that not return an error.

I think this won't make much of a difference internally, since everything's in fx and that'll just translate a panic to an error, but this is definitely one of those "it's a breaking change even though it doesn't break code" things...

Reply:
make sense. I'll go with the start approach first.

Reply:
honestly I'm pretty surprised it didn't return an error earlier.  constructors like this should generally be reserving that for future changes like this...

Reply:
ah, to double check: does all-caps `// DEPRECATED: ...` mark it visibly-deprecated in your IDE?  I think they're gradually getting more flexible about that, but Go uses `// Deprecated: ...` in the standard library and I'm not sure how picky godocs / etc are: https://pkg.go.dev/path/filepath#HasPrefix

Reply:
yea... I think it's case-sensitive: 
<img width="518" alt="Screenshot 2024-10-09 at 18 50 48" src="https://github.com/user-attachments/assets/a1336c1d-b660-4a01-ba8c-8b84ee20a295">

worth changing IMO.  and also probably the indentation on the v2 docs, since I don't think "Returns an error..." is intended to be part of the code block.

Reply:
Changed but local godoc still doesn't show the deprecation warning message. I don't know how to test it.

Reply:
<img width="884" alt="Screenshot 2024-10-10 at 9 58 34 AM" src="https://github.com/user-attachments/assets/9b868ec6-b86e-4cf2-b3da-5bee7e815ca5">
working now after using the suggested tool `https://github.com/golang/pkgsite`

Reply:
yea I have no idea why `godoc` / `go doc` has been seemingly abandoned in the docs changes they've been making :\